procedures: rewrite and expand the "Backup and disaster recovery" sections of the admin guide (#2121)
codenvy/che-docs: modules/administration-guide/partials/con_external-postgresql-setup.adoc

[id="external-postgresql-setup_{context}"]
= External PostgreSQL setup

> By default, {prod-short} Operator creates and manages deployment of the internal PostgreSQL database. However, {prod-short} Operator does not support full life-cycle capabilities such as backups and recovery.

Reviewer: ```suggestion
By default, {prod-short} Operator creates and manages deployment of the internal PostgreSQL database. However, the Operator does not support full life-cycle capabilities such as backups and recovery.
```

> By default, {prod-short} Operator creates and manages deployment of the internal PostgreSQL database. However, {prod-short} Operator does not support full life-cycle capabilities such as backups and recovery.

Reviewer 2: ```suggestion
By default, {prod-short} Operator creates and manages deployment of the internal PostgreSQL database. However, {prod-short} Operator does not support full lifecycle capabilities such as backups and recovery.
```
Per the ISG, `lifecycle` is one word.

Author: Thank you. Fixed.

> An external PostgreSQL database is recommended for a business-critical setup. The external PostgreSQL can be configured for High Availability (HA) and Point In Time Recovery (PITR).

Reviewer 2: ```suggestion
Use an external PostgreSQL database for a business-critical setup and configure it for High Availability (HA) and Point In Time Recovery (PITR).
```
- `PostgreSQL` is a product name that describes a type of database.
-  Per the ISG, avoid recommending actions, and be authoritative: `In most cases,
avoid "recommending" that users do something, especially if the recommendation is vague or unexplained. Instead, describe why the action is problematic or beneficial.`
- Avoid passive voice.

> The external PostgreSQL instance can be configured as on-premises or to use a cloud service such as Amazon Relational Database Service (Amazon RDS). Amazon RDS, for example, enables deployment of production databases in a Multi-Availability Zone configuration for a resilient disaster recovery strategy with daily and on-demand snapshots.

Reviewer 2: ```suggestion
You can configure the external PostgreSQL instance as on premises or to use a cloud service such as Amazon Relational Database Service (Amazon RDS). Using Amazon RDS, for example, you can deploy production databases in a Multi-Availability Zone configuration for a resilient disaster recovery strategy with daily and on-demand snapshots.
```
- Avoid passive voice. Make the sentence user-centric.
- Per ISG: `Do not use. Use "on-premises" or "on-prem" as an adjective before a noun. Otherwise, use "on premises" or "on prem". If capitalized for a product name, write "On-Premises" or "On Premises". See also "on prem".`

> The recommended configuration of an example database:

Reviewer 2: ```suggestion
The following example shows a database configuration for business-critical data:
```
Per ISG and SSG: Avoid `recommend` and instead state the user benefit or the purpose.

[options="header"]
|===
|Parameter | Value
|Instance class | db.t2.small
|vCPU | 1
|RAM | 2 GB
|Multi-az | true, 2 replicas
|Engine version | 9.6.11
|TLS | enabled
|Automated backups | enabled (30 days)
|===

> Making workspace metadata and user information persistent involves configuring the external PostgreSQL database and configuring Che to use the configured external PostgreSQL database:

Reviewer 2: ```suggestion
You can make workspace metadata and user information persistent by configuring the external PostgreSQL database and configuring Che to use the configured external PostgreSQL database:
```
- Re-word to make more user-centric.
- Should `Che` be a variable like `{prod-short}`?

Author: @bburt-rh, thank you and yes.

The {prod-short} server uses a PostgreSQL database for persisting data about the state of {prod-short}. The database contains information about user accounts, workspaces, preferences, and other details.

By default, the {prod-short} Operator creates and manages deployment of this database. However, the {prod-short} Operator does not support full life-cycle capabilities such as backups and recovery.

The following disaster-recovery options are recommended when configuring an external PostgreSQL database for a business-critical setup:

* High Availability (HA)
* Point In Time Recovery (PITR)

The external PostgreSQL instance can be configured as on-premises or to use a cloud service such as Amazon Relational Database Service (Amazon RDS). Amazon RDS enables deployment of production databases in a Multi-Availability Zone configuration for a resilient disaster recovery strategy with daily and on-demand snapshots.

The recommended configuration of an example database:

[options="header"]
|===
|Parameter | Value
|Instance class | db.t2.small
|vCPU | 1
|RAM | 2 GB
|Multi-az | true, 2 replicas
|Engine version | 9.6.11
|TLS | enabled
|Automated backups | enabled (30 days)
|===

> Making workspace metadata and user information persistent involves configuring both the external PostgreSQL and Che:

Reviewer 3: The phrase `external Che` doesn't make sense. We install and use Che. Che has components, one of which is Postgres database. It could be internal (in cluster, managed by us) or external (setup, configured and managed by admin, but just with granted access for Che to store some of its data there).

Author: @mmorhun Just to clarify, I meant here _the external PostgreSQL_ **and** _Che_. (I doubt that I can swap the order in which the following sections appear in the text.)

Author: @mmorhun So what you're saying using internal PostgreSQL database also makes workspace metadata and user information persist, correct? I thought that only using external PostgreSQL database would make workspace metadata and user information persistent. Let me know, I can fix this sentence.

Reviewer 3: @max-cx PostgreSQL database is used to persist users' data. If the database stored in the cluster, it might be deleted when deleting Che. In case of external database server, this is independent process.
